Source/WebCore/editing/EditorCommand.cpp

```diff
@@ -1013,11 +1013,17 @@ static bool executeRemoveFormat(Frame& frame, Event*, EditorCommandSource, const
 
 static bool executeScrollPageBackward(Frame& frame, Event*, EditorCommandSource, const String&)
 {
+    if (frame.settings().eventHandlerDrivenSmoothKeyboardScrollingEnabled() && !frame.eventHandler().focusedScrollableAreaUsesScrollSnap())
+        return frame.eventHandler().keyboardScrollRecursively(ScrollDirection::ScrollUp, ScrollGranularity::Page, nullptr);
+
     return frame.eventHandler().logicalScrollRecursively(ScrollBlockDirectionBackward, ScrollGranularity::Page);
 }
 
 static bool executeScrollPageForward(Frame& frame, Event*, EditorCommandSource, const String&)
 {
+    if (frame.settings().eventHandlerDrivenSmoothKeyboardScrollingEnabled() && !frame.eventHandler().focusedScrollableAreaUsesScrollSnap())
```

Comment:
I wonder if eventHandler should have a getter that checks both of these things, since you have to check both together always? (EventHandler::shouldUseSmoothKeyboardScrollingForFocusedScrollableArea or something?)